cpp/path_module/algorithm/path.cpp

```diff
@@ -301,3 +303,203 @@ void Path::Expand(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result,
+  }
+  value = config.At("relationshipFilter");
+  if (value.IsNull()) {
+    config.Insert("relationshipFilter", mgp::Value(mgp::List()));
+  }
+  value = config.At("labelFilter");
+  if (value.IsNull()) {
+    config.Insert("labelFilter", mgp::Value(mgp::List()));
+  }
+  value = config.At("filterStartNode");
+  if (value.IsNull()) {
+    config.Insert("filterStartNode", mgp::Value(false));
+  }
+  value = config.At("minLevel");
+  if (value.IsNull()) {
+    config.Insert("minLevel", mgp::Value(int64_t(0)));
+  }
+  if (!(config.At("minLevel").IsInt() && config.At("maxLevel").IsInt() && config.At("relationshipFilter").IsList() &&
+        config.At("labelFilter").IsList() && config.At("filterStartNode").IsBool()) ||
+      (config.At("minLevel").ValueInt() != 0 && config.At("minLevel").ValueInt() != 1)) {
+    throw mgp::ValueException(
+        "The config parameter needs to be a map with keys and values in line with the documentation.");
+  }
+}
+
+void Path::SubgraphNodes(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto graph = mgp::Graph(memgraph_graph);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    auto config = arguments[1].ValueMap();
+    SetConfig(config);
+
+    std::unordered_set<mgp::Node> start_nodes;
+    if (arguments[0].IsList()) {
+      for (const auto element : arguments[0].ValueList()) {
+        GetStartNodes(element, graph, start_nodes);
+      }
+    } else {
+      GetStartNodes(arguments[0], graph, start_nodes);
+    }
+
+    LabelSets labelFilterSets;
+    ParseLabels(config.At("labelFilter").ValueList(), labelFilterSets);
+
+    std::map<mgp::Node, std::int64_t> visited_nodes;
+    mgp::List to_be_returned_nodes;
+    for (const auto node : start_nodes) {
+      VisitNode(node, visited_nodes, true, config, 0, labelFilterSets, to_be_returned_nodes);
+    }
+
+    for (auto node : to_be_returned_nodes) {
+      auto record = record_factory.NewRecord();
+      record.Insert(std::string(kResultSubgraphNodes).c_str(), node);
+    }
+
+  } catch (const std::exception &e) {
+    record_factory.SetErrorMessage(e.what());
+    return;
+  }
+}
+
+void Path::SubgraphAll(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto graph = mgp::Graph(memgraph_graph);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    auto config = arguments[1].ValueMap();
+    SetConfig(config);
+
+    std::unordered_set<mgp::Node> start_nodes;
+    if (arguments[0].IsList()) {
+      for (const auto element : arguments[0].ValueList()) {
+        GetStartNodes(element, graph, start_nodes);
+      }
+    } else {
+      GetStartNodes(arguments[0], graph, start_nodes);
+    }
+
+    LabelSets labelFilterSets;
+    ParseLabels(config.At("labelFilter").ValueList(), labelFilterSets);
```

Comment:
Is "labelFilter" always present here?

Reply:
yes, i call   SetConfig()  on line 463 so that I don't have to always check if stuff in config is present